app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -416,81 +432,158 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+            switch (type) {
+                case BottomBarItemAdapter.TYPE_TAB_COUNTER:
+                    chromeViewModel.getShowTabTray().call();
+                    TelemetryWrapper.showTabTrayToolbar();
+                    break;
+                case BottomBarItemAdapter.TYPE_MENU:
+                    chromeViewModel.getShowMenu().call();
+                    TelemetryWrapper.showMenuToolbar();
+                    break;
+                case BottomBarItemAdapter.TYPE_NEW_TAB:
+                    chromeViewModel.getShowNewTab().call();
+                    TelemetryWrapper.clickAddTabToolbar();
+                    break;
+                case BottomBarItemAdapter.TYPE_SEARCH:
+                    chromeViewModel.getShowUrlInput().setValue(getUrl());
+                    TelemetryWrapper.clickToolbarSearch();
+                    break;
+                case BottomBarItemAdapter.TYPE_CAPTURE:
+                    chromeViewModel.getDoScreenshot().call();
+                    // move Telemetry to ScreenCaptureTask doInBackground() cause we need to init category first.
+                    break;
+                case BottomBarItemAdapter.TYPE_PIN_SHORTCUT:
+                    chromeViewModel.getPinShortcut().call();
+                    TelemetryWrapper.clickAddToHome();
+                    break;
+                case BottomBarItemAdapter.TYPE_BOOKMARK:
+                    chromeViewModel.getToggleBookmark().call();
+                    // handling Telemetry in MainActivity
+                    break;
+                case BottomBarItemAdapter.TYPE_REFRESH:
+                    chromeViewModel.getRefreshOrStop().call();
+                    TelemetryWrapper.clickToolbarReload();
+                    break;
+                case BottomBarItemAdapter.TYPE_SHARE:
+                    chromeViewModel.getShare().call();
+                    TelemetryWrapper.clickToolbarShare();
+                    break;
+                case BottomBarItemAdapter.TYPE_NEXT:
+                    chromeViewModel.getGoNext().call();
+                    TelemetryWrapper.clickToolbarForward();
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unhandled menu item in BrowserFragment, type: " + type);
+            }
+        });
+        bottomBar.setOnItemLongClickListener((type, position) -> {
+            if (type == BottomBarItemAdapter.TYPE_MENU) {
+                // Long press menu always show download panel
+                chromeViewModel.getShowDownloadPanel().call();
+                TelemetryWrapper.longPressDownloadIndicator();
+                return true;
+            }
+            return false;
+        });
+        bottomBarItemAdapter = new BottomBarItemAdapter(bottomBar, BottomBarItemAdapter.Theme.LIGHT.INSTANCE);
+        bottomBarViewModel.getItems().observe(this, items -> {
+            bottomBarItemAdapter.setItems(items);
```

Comment:
Since the parameter of Observer#onChanged(value) is marked as nullable, there are a lot of warnings complaining about potential NPE. Maybe we can have something like NonNullLivedata, or NonNullObserver, or just surround the usage with null-check.

Reply:
created the extension function `LiveData<T>.nonNullObserve` to fix the warning